externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala

```diff
@@ -114,9 +112,14 @@ case class SparkSQLEngine(spark: SparkSession) extends Serverable("SparkSQLEngin
         exec,
         Duration(60, TimeUnit.SECONDS))
     })
-    engineSavePath.foreach { p =>
-      val path = new Path(p)
-      path.getFileSystem(spark.sparkContext.hadoopConfiguration).delete(path, true)
+    try {
+      val path = new Path(engineSavePath)
```

Comment:
`if (backendService.sessionManager.getConf.get(OPERATION_RESULT_SAVE_TO_FILE))`

Reply:
as the OPERATION_RESULT_SAVE_TO_FILE can be enable during runtime.

So if we use sessionManager conf to check it here, the engine result save path might be leak.

Reply:
> So if we use sessionManager conf to check it here, the engine result save path might be leak.

Got it. Thanks for the explanation.